src/Microsoft.ApplicationInsights.AspNetCore/DiagnosticListeners/Implementation/HttpHeadersUtilities.cs

```diff
@@ -0,0 +1,112 @@
+namespace Microsoft.ApplicationInsights.AspNetCore.DiagnosticListeners
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+#if NET451
+#endif
+    using System.Net.Http.Headers;
+    using Microsoft.AspNetCore.Http;
+    using Microsoft.Extensions.Primitives;
+#if NETCORE
+    using Microsoft.ApplicationInsights.Common;
+#endif
+    internal static class HttpHeadersUtilities
+    {
+        internal static IEnumerable<string> GetHeaderValues(HttpHeaders headers, string headerName)
```

Comment:
Thanks for the point. For now, I don't see big advantage of extension method and I don't see existing pattern for writing utility as extension methods. Won't fix.
Let me know if you have a strong preference on that.